### DISCUSSION TITLE
Tutorial pipeline works with dagit, errors out with dagster CLI

### DISCUSSION THREAD

**Discussion by gherka** [dagster-io/dagster#3222] [Q&A] [answered]
I'm slowly working through tutorial examples and stumbled across something odd - the `iris_pipeline.py` from this [page](https://docs.dagster.io/overview/packages/dagstermill) works fine when I run it using `dagit -f iris_pipeline.py`, but throws up the following error when I run it using `dagster pipeline execute -f iris_pipeline.py`:
```
dagster.check.CheckError: Failure condition: Can not produce an instance reference for <dagster.core.instance.DagsterInstance object at 0x7fea42d3c290>
```
Is there something obvious that I'm missing?

**Comment by alangenfeld** [comment, marked as answer]
We should catch this with a better error message, but to explain whats happening

The environment `DAGSTER_HOME` controls where dagster will put things on the filesystem, such as marshaling data to the jupyter process for the notebook in the `iris_pipeline`. If this env var is not set, `dagit` will create a temporary directory to use that gets cleaned up when the `dagit` process exits. The `dagster` CLI on the other hand falls back to an in memory "instance", which fails as seen above when trying to prepare the jupyter subprocess. 

You can read more about the "instance" here https://docs.dagster.io/overview/instances/dagster-instance#main

**Reply by alangenfeld** [reply]
error message improvements in https://dagster.phacility.com/D5158
